### ISSUE TITLE
Error on Windows:  Git Error: Error cleaning up cloned template

### ISSUE THREAD

**Issue by gnut-neygun** [cargo-generate/cargo-generate#375]
I got the above error while running:
`cargo generate --git https://github.com/rustwasm/wasm-pack-template.git --name my-project`

Here's the full output: 

>  Creating project called my-project...
> 
> Error:  Git Error: Error cleaning up cloned template

This is on Windows 10, with cargo version 1.53 and rustc 1.53

**Comment by sassman** [comment]
Thanks for the report, I would need few more specific insights.

What shell do you use? 
And from which folder (full path) do you call cargo generate?

**Comment by gnut-neygun** [comment]
I used powershell, cmd, and the built in project wizard provided but Rust plugin in IntelliJ. All same error.
I call it from E:\Experimentation , so normal path with no special character or white spaces.

**Comment by kabirz** [comment]
I have the same problem on windows 10.

**Comment by sassman** [comment]
I was able to reproduce it now and got the following insights:

```
PS Z:\workspaces\rust\cargo-generate> cargo generate --git rustwasm/wasm-pack-template --name demo
 Creating project called `demo`...
[src\git.rs:216] &git_dir = "Z:\\workspaces\\rust\\cargo-generate\\demo\\.git"
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 32, kind: Other, message: "The process cannot access the file because it is being used by another process." }', src\git.rs:218:33
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

so there is for unknown reasons this problem, it happens only on windows so far:

> The process cannot access the file because it is being used by another process.

**Comment by sassman** [comment]
the issue is solved in [v0.7.1](https://github.com/cargo-generate/cargo-generate/releases/tag/v0.7.1).

The problem originated to a windows process blocking error, that happened when the `.git` folder cleanup was triggered. 
At the time of deletion (on windows) still some git related process locks where active. Making sure that all git related structs got dropped before the mentioned delete action is triggered, solved the problem.

**Comment by gnut-neygun** [comment]
That was some incredibly fast fix. Thanks @sassman !
